qiskit_experiments/test/mock_iq_helpers.py

```diff
@@ -0,0 +1,393 @@
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Probability and phase functions for the mock IQ backend."""
+
+from abc import abstractmethod
+from typing import Dict, List
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.providers.aer import AerSimulator
+
+
+class MockIQExperimentHelper:
+    """Abstract class for experiment tools"""
+
+    @abstractmethod
+    def compute_probability(self, circuits: List[QuantumCircuit]) -> List[Dict[str, float]]:
+        """
+        A function provided by the user which is used to determine the probability of each output of the
+         circuit. The function returns a list of dictionaries, each containing output binary strings and
+          their probabilities.
+
+        Examples:
+
+            **1 qubit circuit - excited state**
+
+            In this experiment, we want to bring a qubit to its excited state and measure it.
+            The circuit:
+                         ┌───┐┌─┐
+                      q: ┤ X ├┤M├
+                         └───┘└╥┘
+                    c: 1/══════╩═
+                               0
+
+            The function that calculates the probability for this circuit, doesn't need any
+            calculation_parameters. It will be as following:
+
+            .. code-block::
+
+                @staticmethod
+                def compute_probability() -> List[Dict[str, float]]:
+
+                    output_dict_list = []
+                    for circuit in circuits:
+                        probability_output_dict["1"] = 1
+                        probability_output_dict["0"] = 0
+                        output_dict_list.append(probability_output_dict)
+                    return output_dict_list
+
+            **3 qubit circuit**
+            In this experiment, we prepare a Bell state with the first and second qubit.
+            In addition, we will bring the third qubit to its excited state.
+            The circuit:
+                         ┌───┐     ┌─┐
+                    q_0: ┤ H ├──■──┤M├───
+                         └───┘┌─┴─┐└╥┘┌─┐
+                    q_1: ─────┤ X ├─╫─┤M├
+                         ┌───┐└┬─┬┘ ║ └╥┘
+                    q_2: ┤ X ├─┤M├──╫──╫─
+                         └───┘ └╥┘  ║  ║
+                    c: 3/═══════╩═══╩══╩═
+                                2   0  1
+
+            The backend has a feature, that the output can include only states with probabilities
+            greater than 0. The backend will assume that if an output string isn't in the probability
+            dictionary, it's probability 0.
```

Comment:
```suggestion
            dictionary, its probability is 0.
```

Reply:
I am changing the sentence to:
`When an output string isn't in the probability dictionary, the backend will presume its probability is 0.`